homeassistant/components/webhook/__init__.py

```diff
@@ -87,10 +87,17 @@ def async_generate_id() -> str:
 
 @callback
 @bind_hass
-def async_generate_url(hass: HomeAssistant, webhook_id: str) -> str:
+def async_generate_url(
+    hass: HomeAssistant,
+    webhook_id: str,
+    allow_internal: bool = True,
+    allow_external: bool = True,
+    allow_ip: bool | None = None,
+    prefer_external: bool = False,
```

Comment:
We need to use the same default as in `get_url`:
```suggestion
    refer_external: bool | None = None
```

Let's add a test which specifically checks that the defaults here and in get_url are aligned, the simplest way is probably to use `inspect.signature`

Reply:
done.

Reply:
I undid the change (and the test) because the default value of `prefer_external` should be `True` for backwards compability (and then it doesn't make sense to test the `signature` at all imo)